staging/src/k8s.io/apiserver/pkg/cel/environment/base.go

```diff
@@ -128,6 +128,7 @@ var baseOpts = []VersionedOptions{
 		EnvOptions: []cel.EnvOption{
 			library.IP(),
 			library.CIDR(),
+			library.SemVer(),
```

Comment:
Just wanna mention that adding into base as introduced in 1.30 will not enable the power immediately as we normally would set the `DefaultCompatibilityVersion` one version earlier iff DRA is using the same default env version :)

Reply:
I am solving this over in https://github.com/kubernetes/kubernetes/pull/123516 in `pkg/apis/resource/structured/namedresources/validation/validation.go` like this:
```
func validateSelector(opts Options, selector string, fldPath *field.Path) field.ErrorList {
	var allErrs field.ErrorList
	if selector == "" {
		allErrs = append(allErrs, field.Required(fldPath, ""))
	} else {
		// TODO (https://github.com/kubernetes/kubernetes/issues/123687):
		// when this API gets promoted to beta, we have to
		// validate new and stored expressions differently.
		// While it is alpha, new expressions are allowed to
		// use everything that is currently available.
		// envType := environment.NewExpressions
		// if opts.StoredExpressions {
		//    envType = environment.StoredExpressions
		// }
		envType := environment.StoredExpressions
		result := namedresourcescel.Compiler.CompileCELExpression(selector, envType)
```

Does this look right to you?

Reply:
This would depend on how to build the env in https://github.com/kubernetes/kubernetes/blob/90ed34a12ffc3d58ecedf0e0c91c4d85a67f9aab/staging/src/k8s.io/dynamic-resource-allocation/structured/namedresources/cel/compile.go#L62C19-L62C39.
I saw it's built without version control which means all added func would be available but you'll lose the benefit of version control/backward compatibility. When a new func added in the future, the cluster might running into risk of not recognizing the expression while roll back. I guess it should be fine in Alpha, but that is something better think about going forward in beta/ga. :)

Reply:
True, this needs more thought before proceeding to beta.

Reply:
commented at https://github.com/kubernetes/kubernetes/pull/123516/files#r1516500003 with one possibility for handling this for the namedresources use